Filterable: Add a data-no-filter attribute

carsonip opened the pull request:
Add a data-no-filter attribute for an item such that it will always show in both normal and reveal mode when its data-no-filter="true"

Closes gh-8184

carsonip committed:
Filterable: Add a data-no-filter attribute

Add a data-no-filter attribute for an item such that it will always show in both normal and reveal mode when its data-no-filter="true"

Closes gh-8184

arschmitz commented:
@carsonip can you please sign the CLA?

carsonip committed:
Filterable: data-no-filter attr only accept true

carsonip commented:
@arschmitz I have signed it twice with different names but with the same correct email address. Please check.

arschmitz commented:
@carsonip thank you. We have decided to add this feature though the implementation might be a bit different. Ill review this soon and give you feedback.

jzaefferer commented:
The CLA signature is fine now.

arschmitz commented:
@carsonip sorry about the delay in this finally ready to add this feature. Im wondering if it might be better / more flexible to add an option `ignore` that you can set to any valid selector? thoughts?

apsdehal commented:
I agree with @arschmitz that an option would be more convenient in this case.

carsonip commented:
@arschmitz Sorry for the late reply. I would love to improve the request. Could you elaborate morr about the `ignore` option? Which element should it belong to?

apsdehal commented:
@carsonip ignore option will have a value of string which would a valid html selector for the items which should be not be filtered.

carsonip committed:
Filterable: Use data-ignore instead of data-no-filter

Replaced data-no-filter with data-ignore according to the comments of
jQuery members. Looks more flexible now.

carsonip commented:
Sorry for the wait. I have updated the PR according to the comments. Not sure if there will be a significant performance hit since we will create a jQuery object from each item such that we can use the `is()` function. There will certainly be no performance hit if `data-ignore` is not set.
